src/test/java/com/mael/payroll/paymentMethods/MailMethodTest.java

```diff
@@ -0,0 +1,14 @@
+package com.mael.payroll.paymentMethods;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MailMethodTest {
+
+    @Test
+    public void createsAMailMethod() {
```

Comment:
Would something like `hasAnAddress` fit better here?
In the end you're not verifying the creation of a mail method here, are you?